CRM/Core/Smarty/plugins/modifier.crmDate.php

```diff
@@ -21,13 +21,35 @@
  * @param string $dateString
  *   Date which needs to converted to human readable format.
  *
- * @param null $dateFormat
+ * @param string|null $dateFormat
+ *   A string per https://www.php.net/manual/en/function.strftime.php or
+ *   one of our configured formats name - eg
+ *    - dateformatDatetime
+ *    - dateformatFull
+ *    - dateformatPartial
+ *    - dateformatTime
+ *    - dateformatYear
+ *    - dateformatFinancialBatch
+ *    - dateformatshortdate
+ *
  * @param bool $onlyTime
  *
  * @return string
  *   human readable date format | invalid date message
  */
-function smarty_modifier_crmDate($dateString, $dateFormat = NULL, $onlyTime = FALSE) {
+function smarty_modifier_crmDate(string $dateString, ?string $dateFormat = NULL, bool $onlyTime = FALSE): string {
+  $configuredFormats = [
+    'dateformatDatetime',
+    'dateformatFull',
+    'dateformatPartial',
+    'dateformatTime',
+    'dateformatYear',
+    'dateformatFinancialBatch',
+    'dateformatshortdate',
+  ];
+  if (in_array($dateFormat, $configuredFormats, TRUE)) {
+    $dateFormat = Civi::settings()->get($dateFormat);
```

Comment:
Yeah, I like the concept. 

This may be a little bikesheddy, but `{$foo_date|crmDate:"dateformatDatetime"}` seems to have the word "date" a lot. Maybe skip the "dateformat" bit? eg

```php
  $configuredFormats = ['Datetime', 'Full', 'Partial', 'Time', 'Year', 'FinancialBatch', 'shortdate'];
  if (in_array($dateFormat, $configuredFormats, TRUE)) {
    $dateFormat = Civi::settings()->get('dateformat' . $dateFormat);
  }
```